lib/view_component/base.rb

```diff
@@ -180,7 +180,7 @@ def compile!
       # We could in theory do this on app boot, at least in production environments.
       # Right now this just compiles the first time the component is rendered.
       def compile(raise_template_errors: false)
-        return if compiled? || inlined?
+        return if compiled? || all_inlined?
```

Comment:
What do you think about refactoring this logic to just use `compiled?` This process should only run once per component anyways.